script/spec-runner.js

```diff
@@ -0,0 +1,81 @@
+const cp = require('child_process')
+const crypto = require('crypto')
+const fs = require('fs')
+const { hashElement } = require('folder-hash')
+const path = require('path')
+
+const BASE = path.resolve(__dirname, '../..')
+const NPM_CMD = process.platform === 'win32' ? 'npm.cmd' : 'npm'
+
+const specHashPath = path.resolve(__dirname, '../spec/.hash')
+if (!fs.existsSync(specHashPath)) {
+  fs.writeFileSync(specHashPath, 'invalid\ninvalid')
+}
+
+const [lastSpecHash, lastSpecInstallHash] = fs.readFileSync(specHashPath, 'utf8').split('\n')
+getSpecHash().then(([currentSpecHash, currentSpecInstallHash]) => {
+  const specChanged = currentSpecHash !== lastSpecHash
+  const installChanged = lastSpecInstallHash !== currentSpecInstallHash
+  if (specChanged || installChanged) {
+    const out = cp.spawnSync(NPM_CMD, ['install'], {
+      env: Object.assign({}, process.env, {
+        npm_config_nodedir: path.resolve(BASE, 'out/Default/gen/node_headers'),
+        npm_config_msvs_version: '2017'
```

Comment:
Shouldn't this only be set for Windows?

Reply:
@jkleinsc Setting it on non-windows platform is a noop.  MSVS only impacts win32 module rebuilding